modules/exploits/linux/http/opennms_horizon_authenticated_rce.rb

```diff
@@ -0,0 +1,932 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  prepend Msf::Exploit::Remote::AutoCheck
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'OpenNMS Horizon Authenticated RCE',
+        'Description' => %q{
+          This module exploits built-in functionality in OpenNMS
+          Horizon in order to execute arbitrary commands as the
+          opennms user. For versions 32.0.2 and higher, this
+          module requires valid credentials for a user with
+          ROLE_FILESYSTEM_EDITOR privileges and either
+          ROLE_ADMIN or ROLE_REST.
+
+          For versions 32.0.1 and lower, credentials are
+          required for a user with ROLE_FILESYSTEM_EDITOR,
+          ROLE_REST, and/or ROLE_ADMIN privileges. In that case,
+          the module will automatically escalate privileges via
+          CVE-2023-40315 or CVE-2023-0872 if necessary.
+
+          This module has been successfully tested against OpenNMS
+          version 31.0.7
```

Comment:
Should this be 31.0.8? The low bound listed on NVD seems to be 31.0.8/ that's also the version listed in docker compose file in the docs. 
```suggestion
          version 31.0.8
```